packages/ui/src/components/UnitInput/index.tsx

```diff
@@ -0,0 +1,329 @@
+    }
+
+    return 'neutral'
+  }, [error, success])
+
+  useEffect(() => {
+    const selectBar = document.getElementById('select-bar')
+
+    if (selectBar) {
+      selectBar.style.backgroundColor = 'transparent'
+      selectBar.style.border = 'none'
+      selectBar.style.borderRadius = '0px'
+    }
+  }, [])
+
+  return (
+    <Stack gap={0.5}>
+      {label ? (
+        <Stack direction="row" gap={0.5} alignItems="center">
+          {label ? (
+            <Text
+              as="label"
+              variant={size === 'large' ? 'bodyStrong' : 'bodySmallStrong'}
+              disabled={disabled}
+            >
+              {label}
+            </Text>
+          ) : null}
+          {required && label ? (
+            <Icon name="asterisk" sentiment="danger" size={8} />
+          ) : null}
+          {labelInformation && label ? labelInformation : null}
+        </Stack>
+      ) : null}
+      <UnitInputWrapper
+        direction="row"
+        data-testid={dataTestId}
+        data-size={size}
+        data-focus={hasFocus}
+        width={width}
+        id={id}
+        data-success={!!success}
+        data-error={!!error}
+        onMouseEnter={() =>
+          !disabled && !readOnly ? setHovered(true) : undefined
+        }
+        onMouseLeave={() => setHovered(false)}
+        data-disabled={!!disabled}
+        data-readonly={!!readOnly}
+      >
+        <StyledInputWrapper id="input-field" data-hovered={hovered}>
+          <StyledInput
+            type="number"
+            aria-invalid={!!error}
+            autoFocus={autoFocus}
+            disabled={disabled}
+            data-readOnly={readOnly}
+            name={`${name}-value`}
+            value={val}
+            onChange={event => {
+              const numericValue = Number.parseInt(event.target.value, 10)
+              if (numericValue > maxValue) {
+                setVal(maxValue)
+                onChange?.(maxValue)
+              } else if (numericValue < minValue) {
+                setVal(minValue)
+                onChange?.(minValue)
+              } else {
+                setVal(numericValue)
+                onChange?.(numericValue)
+              }
+            }}
+            placeholder={placeholder}
+            max={maxValue}
+            readOnly={readOnly}
+            min={minValue}
+            step={step}
+            data-success={success}
+            data-error={error}
+            data-testid="unit-input"
+            required={required}
+            className={className}
+            onFocus={() =>
+              !disabled && !readOnly ? setHasFocus(true) : undefined
+            }
+          />
+          {error ? <Icon name="alert" sentiment="danger" /> : null}
+          {success && !error ? (
+            <Icon name="checkbox-circle-outline" sentiment="success" />
+          ) : null}
+        </StyledInputWrapper>
+        <CustomSelectInput
+          width={selectInputWidth}
+          id={`${name}-unit`}
```

Comment:
The id should be based using `id` provided in the parent if it is provided:

```tsx
id={id ? `${id}-unit` : undefined}
```

And the StyledInput too should have an id then